mysite/psa/pipeline.py

```diff
@@ -0,0 +1,296 @@
+from django.template import RequestContext
+from django.shortcuts import render_to_response
+from django.contrib.auth import login, logout
+# from django.core.mail import send_mail
+from django.conf import settings
+from django.db import IntegrityError
+from django.contrib.auth.models import User
+import time
+from datetime import datetime
+
+from social.pipeline.partial import partial
+from social.exceptions import (InvalidEmail,
+                               AuthException,
+                               AuthAlreadyAssociated)
+from social.backends.utils import load_backends
+from social.apps.django_app.default.models import UserSocialAuth
+
+from psa.models import AnonymEmail, SecondaryEmail
+
+
+# @partial
+# def password_ask(strategy, details, user=None, is_new=False, *args, **kwargs):
+# if is_new and kwargs.get('backend').name == 'email':
+#         email = user.email or details.get('email')
```

Comment:
> What is this?

Partial pipeline for on-the-fly password asking. Now we do not use this pipeline.